google/resource_google_project.go

```diff
@@ -155,6 +155,13 @@ func resourceGoogleProjectRead(d *schema.ResourceData, meta interface{}) error {
 		return handleNotFoundError(err, d, fmt.Sprintf("Project %q", pid))
 	}
 
+	// If the project has been deleted from outside Terraform, remove it from state file.
+	if p.LifecycleState != "ACTIVE" {
```

Comment:
Note:

LifecycleState has the following possible values:
- "ACTIVE"
- "DELETE_REQUESTED"
- "DELETE_IN_PROGRESS"